MsfinderCommon/Process/PeakAssigner.cs

```diff
@@ -44,11 +44,17 @@ public sealed class PeakAssigner
             var curatedPeaklist = getCuratedPeaklist(formulaResult.ProductIonResult);
             
             var results = MainProcess.Fragmenter(eQueries, rawData, curatedPeaklist, refinedPeaklist, adductIon, formulaResult, param, fragmentDB, fragmentOntologies);
-
+            
             foreach (var result in results) {
                 result.TotalScore += formulaResult.TotalScore;
                 result.Ontology = rawData.Ontology;
             }
+            if (results.Count == 0)
+            {
+            // If the results is 0, write to a file in SMI format
+                var logSmileFilePath = FileStorageUtility.GetSmileLogFilePath(analysisFile.FormulaFilePath);
+                FileStorageUtility.WriteToSmiFile(rawData.Smiles, logSmileFilePath);
+            }

```

Comment:
So this means overall if the fragmenter didn't find any results? Is that also the case if everything is parsed correctly, just there are no peak annotations found?